logger/mocklogger_test.go

```diff
@@ -0,0 +1,67 @@
+package logger
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	router "gopkg.in/Clever/kayvee-go.v5/router"
+)
+
+func TestMockLoggerImplementsKayveeLogger(t *testing.T) {
+	assert.Implements(t, (*KayveeLogger)(nil), &MockRouteCountLogger{}, "*MockRouteCountLogger should implement KayveeLogger")
+}
+
+func TestRouteCountsWithMockLogger(t *testing.T) {
+	routes := map[string](router.Rule){
+		"rule-one": router.Rule{
+			Matchers: router.RuleMatchers{
+				"foo": []string{"bar", "baz"},
+			},
+			Output: router.RuleOutput{
+				"out": "#-%{foo}-",
+			},
+		},
+		"rule-two": router.Rule{
+			Matchers: router.RuleMatchers{
+				"abc": []string{"def"},
+			},
+			Output: router.RuleOutput{
+				"more": "x",
+			},
+		},
+	}
+	testRouter := router.NewFromRoutes(routes)
+
+	mockLogger := NewMockCountLogger("testing")
```

Comment:
I was thinking users would do something like:

1. implement a method (`MyRouteHandler()`)
2. in this method, do some kayvee logging
3. write a test which checks -- when `MyRouteHandler()` gets hit, does log L get to destination D (say, Redshift) via Rule R? 

Ideally, people shouldn't need to care much at all about `NewMockCountLogger()`, except as 1-line of setup in tests. It might look like:

1. `NewRouter(l *Logger, ...)` -- set a mock logger in tests. This seems like the best way, but requires a stricter app design with dependency injection of the logger
2. override: `router.Logger = NewMockCountLogger()` -- would need to do this in a way that "just works", so it reads in the routes file / is configured identically to the real logger in the app.

Does that mesh with what you were thinking?